front/lib/api/auth0.ts

```diff
@@ -15,3 +24,80 @@ export function getAuth0ManagemementClient(): ManagementClient {
 
   return auth0ManagemementClient;
 }
+
+/**
+ * Verify an Auth0 token.
+ * Not meant to be exported, use `getUserFromAuth0Token` instead.
+ */
+export function verifyAuth0Token(idToken: string): Promise<jwt.JwtPayload> {
+  const client = jwksClient({
+    jwksUri: AUTH0_VERIFY_TOKEN,
+    cache: true,
+    rateLimit: true,
+  });
+
+  function getKey(header: jwt.JwtHeader, callback: jwt.SigningKeyCallback) {
+    client.getSigningKey(header.kid!, (err, key) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+      if (!key) {
+        callback(new Error("Key not found"));
+        return;
+      }
+      const signingKey = key.getPublicKey();
+      callback(null, signingKey);
+    });
+  }
+
+  return new Promise((resolve, reject) => {
+    jwt.verify(
+      idToken,
+      getKey,
+      {
+        algorithms: ["RS256"],
+        audience: AUTH0_AUDIENCE,
+        issuer: AUTH0_ISSUER,
+      },
+      (err, decoded) => {
+        if (err) {
+          reject(err);
```

Comment:
`nit` to help parsing code.
```suggestion
          return reject(err);
```